docs/reactive-nonblocking.qmd

```diff
@@ -0,0 +1,316 @@
+2. Perform the task asynchronously and concurrently, outside of the reactive graph.
+3. When the task completes, bring the resulting value (or error, if the operation failed) back into the reactive graph.
+
+We've created a high-level class called `ExtendedTask` to make all of this pretty easy.
+To create an `ExtendedTask`, use the `@reactive.extended_task` decorator on an async function.
+Let's adapt the example above to use an `ExtendedTask`:
+
+```{shinylive-python}
+#| standalone: true
+#| components: [editor, viewer]
+import asyncio
+import datetime
+from random import random
+import time
+from shiny import App, reactive, render, ui
+
+app_ui = ui.page_fluid(
+    ui.p("The time is ", ui.output_text("current_time", inline=True)),
+    ui.hr(),
+    ui.input_numeric("x", "x", value=1),
+    ui.input_numeric("y", "y", value=2),
+    ui.input_task_button("btn", "Add numbers"),
+    ui.output_text("sum"),
+)
+
+def server(input, output, session):
+    @render.text
+    def current_time():
+        reactive.invalidate_later(1)
+        return datetime.datetime.now().strftime("%H:%M:%S %p")
+
+    @ui.bind_task_button(button_id="btn")
+    @reactive.extended_task
+    async def sum_values(x, y):
+        await asyncio.sleep(5)
+        return x + y
+
+    @reactive.effect
+    @reactive.event(input.btn)
+    def btn_click():
+        sum_values(input.x(), input.y())
+
+    @render.text
+    def sum():
+        return str(sum_values.result())
+
+app = App(app_ui, server)
+```
+
+Note the `sum_values` function, which is where the actual (slow) work is done.
+It is decorated with `@reactive.extended_task`, which means that it will be run asynchronously and concurrently with other tasks, and that its result will be available as `sum_values.result()`.
+
+We've also added a `@ui.bind_task_button` decorator on top of the `@reactive.extended_task` decorator.
+This synchronizes the `ExtendedTask` object with the `ui.input_task_button` in the UI, so that the button will be in its "Processing..." state while the task is running.
+It _does not_ cause button clicks to invoke the task; we still need to do that manually, which we'll talk about next.
+
+If you use some other UI gesture or condition besides `ui.input_task_button` to invoke the task, don't include the `@ui.bind_task_button` decorator--it doesn't work with `ui.input_action_button`, for example.
+
+### Invoking an extended task
+
+Unlike a reactive effect, simply creating an extended task does not cause it to run.
+It needs to be invoked (called like a function).
+
+In this case, the `sum_values` extended task is called from the `btn_click` reactive effect (`@reactive.effect`), which runs whenever the button is clicked (`@reactive.event(input.btn)`).
+
+Notice also that the `sum_values` logic no longer reads `input.x()` and `input.y()` directly; as an extended task, it is not allowed to read reactive inputs or calcs (and attempting to do so would result in an error).
```

Comment:
```suggestion
Notice also that the `sum_values` logic no longer reads `input.x()` and `input.y()` directly in the function body (doing so would lead to an error). 
Instead, when invoking the task, we pass reactive values to it.
```